### ISSUE TITLE
ylim exaggerated

### ISSUE THREAD

**Issue by dpo** [Evizero/UnicodePlots.jl#57]
Not sure if this is the same as #21 but setting `ylim=[0, 1.1]` actually ends up using `[0, 2]`, which seems a bit overboard:

```
julia> using UnicodePlots
julia> lineplot(collect(1:100), rand(100), ylim=[0, 1.1])
     ┌────────────────────────────────────────┐
   2 │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⡄⢠⠀⠀⠀⢠⡄⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀⠀│
     │⢠⡄⠀⢀⠀⠀⡄⠀⠀⠀⡀⠀⠀⠀⢸⠀⡇⣿⣧⠀⠀⢸⣧⠀⠀⠀⠀⣷⠀⢠⠀⠀⢰⠀⠀⡆⠀⠀⢠⠀│
     │⢸⢇⢰⣼⡇⣾⣇⢀⠀⢸⡇⠀⠀⠀⢸⡀⣷⣿⣿⠀⠀⢸⣿⡄⠀⠀⢀⢿⢀⣼⣇⢀⣼⠀⡆⣧⡆⢰⣾⠀│
     │⢸⠘⡾⣿⣷⣿⣿⣸⠀⢸⡇⡆⠀⡀⢸⡇⣿⣿⣿⢰⡇⢸⠇⡇⠀⢀⣼⢸⢸⣿⣿⢸⡏⡆⣿⣿⢣⡸⣿⠀│
     │⡜⠀⠁⢿⣿⢿⣿⡏⡆⢸⡇⣷⣷⣧⣼⣷⢻⢿⣿⢸⢇⢸⠀⡇⡄⢸⣿⢸⡎⠏⢻⢸⠀⣿⠸⣿⢸⡇⡿⣦│
     │⠃⠀⠀⢸⣿⢸⣿⡇⡇⢸⣧⢻⠁⠘⡟⢿⢸⠸⠇⣿⢸⣿⠀⢣⣿⡎⡿⢸⡇⠀⠸⡇⠀⣿⠀⢸⢸⡇⡇⡇│
     │⠀⠀⠀⢸⣿⠸⡇⠁⡇⢸⣿⠈⠀⠀⡇⠈⢸⠀⠀⣿⠀⣿⠀⢸⡏⠃⠁⠸⡇⠀⠀⡇⠀⠹⠀⠈⢸⠇⠀⡇│
   0 │⠀⠀⠀⠘⠇⠀⠀⠀⠉⠋⠸⠀⠀⠀⠀⠀⠀⠀⠀⢿⠀⢹⠀⠘⠃⠀⠀⠀⠀⠀⠀⠁⠀⠀⠀⠀⠀⠀⠀⠀│
     └────────────────────────────────────────┘
     0                                      100
```

Testing other values of `ylim` it seems the upper bound is rounded up to the next integer. Is that the case?

**Comment by Evizero** [comment]
Yes, it could be the case that there are quite strange roundings at places. Turns out sensible rounding of plot limits is not trivial and my hack may be a bit conservative. I have seen a while back that @tbreloff created a PlotUtils package that has some tick logic in them which may be useful for this (?). I may turn to that in my planned refactor

**Comment by Evizero** [comment]
fixed in #85
